packages/react/src/utils/factories.ts

```diff
@@ -250,10 +250,17 @@ function createShorthandFromValue<P>({
 
   // Merge styles
   if (defaultProps.styles || overrideProps.styles || usersProps.styles) {
-    ;(props as any).styles = mergeStyles(
-      defaultProps.styles,
-      usersProps.styles,
-      overrideProps.styles,
+    // As long as we allow the styles prop to be a function, I don't see how we can cache it, as the stirng of the function will always be mergeStyles..
+    // ;(props as any).styles = mergeStyles(
+    //   defaultProps.styles,
+    //   usersProps.styles,
+    //   overrideProps.styles,
+    // )
+    // Just for hacking, relaying that it will always be an object, so we can cache...
+    ;(props as any).styles = _.merge(
```

Comment:
@levithomason clarification, if I use here mergeStyles function, the result in JSON.stringify is always the same function string, that's why it fails... I will think of how this can be fixed